### PR TITLE
KT-4519 Show first line with content of block comment when collapsed

### PR DESCRIPTION
**Pull request by shiraji**
#KT-4519 Fixed

https://youtrack.jetbrains.com/issue/KT-4519

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt [hunk 1]
```diff
@@ -97,12 +97,28 @@ class KotlinFoldingBuilder : CustomFoldingBuilder(), DumbAware {
...
 
+    private fun getFirstLineOfComment(node: ASTNode): String {
+        val targetCommentLine = node.text.split("\n").firstOrNull {
+            getCommentContents(it).isNotEmpty()
+        } ?: return ""
+        return getCommentContents(targetCommentLine)
+    }
+
+    private fun getCommentContents(line: String): String {
+        val comment = KotlinCommenter()
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: There is no need to create a new object here; it's OK to hard-code the token texts.

File: idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt [hunk 2]
```diff
@@ -97,12 +97,28 @@ class KotlinFoldingBuilder : CustomFoldingBuilder(), DumbAware {
...
+        return line.trim()
+                .replace(comment.documentationCommentPrefix!!, "")
+                .replace(comment.blockCommentPrefix!!, "")
+                .replace(comment.blockCommentSuffix!!, "")
+                .replace(comment.documentationCommentLinePrefix!!, "")
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: Please make sure that the line has at most one leading and at most one trailing space after you've remove the prefix and suffix

Comment (shiraji) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: How about trimming both of spaces here and add one leading/trailing space at https://github.com/JetBrains/kotlin/pull/928/files#diff-e4fc5e886f7f1903e7f5f67545e46564R110

**Comment by yole** [comment]
Thanks a lot for your contribution!

**Comment by shiraji** [comment]
Thanks!

**Pull request merged**